Fix undo/redo being locked up when a MouseUp event occurs while an object is being dragged

chris-ehmann opened the pull request:
Partially addresses #32543, specifically "Ctrl + Z doesn't bring the deleted object back" (since I believe they're just accidentally middle-clicking while scrolling)

When you quick-delete an object in the editor while it's being dragged (either through a middle mouse button click or shift + right click), it causes the entire undo/redo system to break. I believe that this is caused by `changeHandler` never being able to call `EndChange`, as `isDraggingBlueprint` is erroneously set to false by the middle (or right) mouse button click.

We can fix this with a simple check to see if the MouseUpEvent was specifically caused by the left mouse button.

_quick edit: im realizing that this isn't strictly related to objects being deleted, and undo/redo will be locked up when any sort of non left-click mouseup event occurs while an object is being dragged, so i'm renaming to be a bit more appropriate._

# Video

## Before

https://github.com/user-attachments/assets/129d0008-7c8e-47cf-a0c3-97fed1c68f9d

## After

https://github.com/user-attachments/assets/e0de121d-ce35-414e-899c-58b75cff237a

chris-ehmann committed:
Add check to see if MouseUp event was the left button

bdach reviewed (changes requested):
I haven't managed to find any breakage here but this 200% needs to be covered by tests because editor input handling code is horribly complex and I'm not going to allow guards with non-obvious purpose without documenting what they're for, in the form of inline comments and/or test coverage.

Test should live in `osu.Game.Tests.Visual.Editing.TestSceneComposerSelection`.

chris-ehmann committed:
Add test for undoing after quick deleting an object while it is dragged.

chris-ehmann committed:
Add comment to explain guard clause

chris-ehmann committed:
better comment, fix typo in test case

chris-ehmann commented:
I've added a comment explaining the purpose of the guard clause and test coverage covering the process of dragging an object -> delete via middle click -> undo. I assume this should be good since other processes of testing right click instead or other mouse clicks instead would essentially be repeating the same test, unless I'm completely overlooking something.

bdach committed:
Fix up test

- Use constraints for better assert messages
- Use `Editor.Undo()` rather than manual input manager synthesizing
  ctrl-z (ctrl-z is not a platform agnostic binding, see macOS)

bdach committed:
Merge branch 'master' into fix-drag-delete-bug
